components/templates/templates.js

```diff
@@ -330,5 +337,27 @@ if (!utils.templates) {
         target.appendChild(templates.item(c));
       }
     };
+
+    if (document.registerElement) {
+      var xTemplateProto = Object.create(HTMLTemplateElement.prototype);
+
+      // When an 'x-template' is created the element in the DOM is destroyed and
+      // moved to the Shadow DOM
+      xTemplateProto.createdCallback = function() {
```

Comment:
you are totally right. I have added a comment to the comment at the beginning of the library